charts/customer-center/templates/secret-backend.yaml

```diff
@@ -0,0 +1,22 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ include "customerCenter.fullname" . }}-backend
+  labels:
+    {{- include "customerCenter.labels" . | nindent 4 }}
+    app.kubernetes.io/component: backend
+type: Opaque
+data:
+  IDP_CLIENT_SECRET: {{ .Values.backend.settings.idp.clientSecret | b64enc | quote }}
+  {{ if .Values.postgresql.postgresqlPassword }}
+  POSTGRES_PASSWORD: {{ .Values.postgresql.postgresqlPassword | b64enc | quote }}
+  {{ else }}
+  POSTGRES_PASSWORD: {{ randAlphaNum 32 | b64enc | quote }}
+  {{ end }}
+  AUTH_SECRET: {{ .Values.backend.settings.auth.secret | b64enc | quote }}
+  LDAP_BIND_CREDENTIALS: {{ .Values.backend.settings.ldap.bindCredentials | b64enc | quote }}
+  {{ if .Values.redis.password }}
```

Comment:
@eyenx Don't we need leading `-` within the `if `and `else` statements for a proper templating without empty lines?

Reply:
@vmaillot this needs fixing. If I template the chart, the secret comes out like this.

```yaml
# Source: customer-center/templates/secret-backend.yaml
apiVersion: v1
kind: Secret
metadata:
  name: RELEASE-NAME-customer-center-backend
  labels:
    helm.sh/chart: customer-center-0.1.0
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/part-of: customer-center
    app.kubernetes.io/version: "2.0.0-rc.4"
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: customer-center
    app.kubernetes.io/instance: RELEASE-NAME
    app.kubernetes.io/component: backend
type: Opaque
data:
  IDP_CLIENT_SECRET: "dGhpcy1pcy1teS1jbGllbnQtc2VjcmV0"

  POSTGRES_PASSWORD: "cGFzc3dvcmQ="

  AUTH_SECRET: "bXktc3VwZXItc2VjcmV0"
  LDAP_BIND_CREDENTIALS: "YmluZC1jcmVkZW50aWFscw=="

  REDIS_PASSWORD: "bXktcmVkaXMtcGFzc3dvcmQ="
```

Thanks for finding this @sinicis

Reply:
> First, the curly brace syntax of template declarations can be modified with special characters to tell the template engine to chomp whitespace. {{- (with the dash and space added) indicates that whitespace should be chomped left, while -}} means whitespace to the right should be consumed. Be careful! Newlines are whitespace!

Taken from: https://helm.sh/docs/chart_template_guide/control_structures/